src/App.js

```diff
@@ -400,7 +401,24 @@ class _AppStackWrapper extends React.Component {
 
       if (!this.props.isRecoveringPin) {
         if (this.props.isScreenLocked) {
-          screen = <PinScreen isLockScreen navigation={this.props.navigation} />;
+          /**
+           * NOTE:
+           * This approach shows the ResetWallet screen as an auxiliar view,
+           * replacing the PinScreen and setting the back button to drop the view,
+           * letting the PinScreen be re-rendered.
+           *
+           * This approach also keeps the navigation stack unchanged,
+           * therefore increasing the convinience for the user.
+           *
+           * The previous approach unlocks the wallet and navigates to ResetWallet
+           * screen with a back button pointing to StartWalletLockScreen,
+           * this interfering in the stack navigation.
```

Comment:
I don't think we should have this part of the comment here, just add it as a comment on the PR, future developers shouldn't care about the "previous approach"

Reply:
Resolvido em [716ecaf](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/261/commits/716ecaf1c32fe99dc9375120061ee873019e20e1)